File: bronze/2798/python/main.py

if (__name__ == "__main__"):
    n, m = map(int, input().split())
    card = list(map(int, input().split()))[:n]
    _max = 0
    
    for i in range(n):
        for j in range(i+1, n):
            _sum = card[i] + card[j]
            if (_sum >= m): continue

            for k in range(j+1, n):
                _sum = card[i] + card[j] + card[k]
                if (_max < _sum and _sum <= m):
                    _max = _sum
    print(_max)

File: bronze/2798/c++/main.cpp
#include <iostream>

int main()
{
    int n, m;
    int sum, max = 0;

    std::cin >> n >> m;
    int card[n];
    
    for (int i = 0 ; i < n; i++) {
        std::cin >> card[i];
    }

    for (int i = 0; i < n; i++) {
        for (int j = i +1; j < n; j++) {
            sum = card[i] + card[j];
            if (sum >= m) { continue; }
            for (int k = j +1; k < n; k++) {
                sum = card[i] + card[j] + card[k];
                if (max < sum && sum <= m) {
                    max = sum;
                }
            }
        }
    }
    std::cout << max;


    return 0;
}